modules/backend/traits/ModelOptions.php

```diff
@@ -0,0 +1,92 @@
+<?php namespace Backend\Traits;
+
+/**
+ * Model Options Trait
+ * Implements helper methods useful to extract the options from a given field
+ * that is used in the List and Form widgets.
+ */
+
+trait ModelOptions
+{
+    /**
+     * Looks at the model for defined options.
+     *
+     * @param $field
+     * @param $fieldOptions
+     * @return mixed
+     */
+    protected function getOptionsFromModel($field, $fieldOptions, $fieldName = '')
+    {
+        /*
+         * Advanced usage, supplied options are callable
+         */
+        if (is_array($fieldOptions) && is_callable($fieldOptions)) {
+            $fieldOptions = call_user_func($fieldOptions, $this, $field);
+        }
+        /*
+         * Refer to the model method or any of its behaviors
+         */
+        if (!is_array($fieldOptions) && !$fieldOptions) {
+            if(empty($fieldName)) {
+                try {
+                    list($model, $attribute) = $field->resolveModelAttribute($this->model, $field->fieldName);
+                }
+                catch (Exception $ex) {
+                    throw new ApplicationException(Lang::get('backend::lang.field.options_method_invalid_model', [
+                        'model' => get_class($model),
+                        'field' => $field->fieldName
+                    ]));
+                }
+            } else {
+                $methodName = 'get'.studly_case($fieldName).'Options';
```

Comment:
@SaifurRahmanMohsin Please add spaces around the full stops.
```suggestion
                $methodName = 'get' . studly_case($fieldName) . 'Options';
```